pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTaskRestletResource.java

```diff
@@ -632,8 +632,9 @@ public Map<String, String> scheduleTasks(@ApiParam(value = "Task type") @QueryPa
       Map<String, List<String>> allTaskNames = tableName != null ? _pinotTaskManager.scheduleAllTasksForTable(
           DatabaseUtils.translateTableName(tableName, headers), minionInstanceTag)
           : _pinotTaskManager.scheduleAllTasksForDatabase(database, minionInstanceTag);
-      return allTaskNames.entrySet().stream()
+      Map<String, String> result = allTaskNames.entrySet().stream().filter(entry -> entry.getValue() != null)
           .collect(Collectors.toMap(Map.Entry::getKey, entry -> String.join(",", entry.getValue())));
+      return result.isEmpty() ? null : result;
```

Comment:
can we mark the API with `@Nullable` annotation? Also not introduced in this PR but can we also add `@Nullable` annotation to taskType and tableName.

Reply:
Nice catch. Thank you!
Added the annotations.